.github/workflows/esy-build.yml

```diff
@@ -3,26 +3,28 @@ name: Build using esy
 on:
   pull_request:
   push:
-   branches:
-     - fork
+    branches:
+      - fork
 
 jobs:
   build:
     name: Build
 
     strategy:
       matrix:
-        system: [windows-latest, ubuntu-latest, macos-latest]
+        system: [windows, ubuntu, macos, macos-arm64]
 
-    runs-on: ${{ matrix.system }}
+    runs-on: ${{ matrix.system }}-latest
 
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
+        if: ${{ matrix.system != 'macos-arm64' }}
         with:
           node-version: 14
 
       - name: Install esy
+        if: ${{ matrix.system != 'macos-arm64' }}
```

Comment:
Because the github actions environment is still running on rosetta, when we try to install node on the internal environment it installs npm x86_64 which will install esy x86_64.

So to avoid that we have esy and npm preinstalled on the self-hosted computer